// @hey-api/openapi-ts/dist/index.cjs

const require_openApi = require('./openApi-BetJIgjU.cjs');
const require_src = require('./src-CJqsBn3S.cjs');

exports.$ = require_openApi.$;
exports.Logger = require_src.Logger;
exports.OperationPath = require_openApi.OperationPath;
exports.OperationStrategy = require_openApi.OperationStrategy;
exports.TsDsl = require_openApi.TsDsl;
exports.TsDslContext = require_openApi.TsDslContext;
exports.TypeScriptRenderer = require_openApi.TypeScriptRenderer;
exports.clientDefaultConfig = require_openApi.clientDefaultConfig;
exports.clientDefaultMeta = require_openApi.clientDefaultMeta;
exports.clientPluginHandler = require_openApi.clientPluginHandler;
exports.createClient = require_src.createClient;
exports.ctx = require_openApi.ctx;
exports.defaultPaginationKeywords = require_openApi.defaultPaginationKeywords;
exports.defaultPlugins = require_openApi.defaultPlugins;
exports.defineConfig = require_src.defineConfig;
exports.definePluginConfig = require_openApi.definePluginConfig;
exports.keywords = require_openApi.keywords;
exports.regexp = require_openApi.regexp;
exports.reserved = require_openApi.reserved;
exports.utils = require_src.utils;
